extendr-api/src/optional/either.rs

```diff
@@ -1,5 +1,36 @@
+/*!
+Enables support for [`either`](https://docs.rs/either/latest/either/) crate, allowing accepting and returning `Either<L,R>` values if both `L` and `R` are convertible to/from `Robj`.
```

Comment:
```suggestion
Enables support for the [`either`](https://docs.rs/either/latest/either/) crate, to allow accepting and returning `Either<L, R>` values if both `L` and `R` are convertible to/from `Robj`.
```

Reply:
I am not convinced '*to allow*' is grammatically correct here. To me, this reads '*enables smth, to allow smth*', while I'd expect either '*enable smth, allowing smth*' or '*enables smth and allows smth*'.

Reply:
> To me, this reads 'enables smth, to allow smth'

This sounds fine to me. It's short for "This module enable**s** support for the either crate, to allow...". Does that sound better to you?

> 'enable smth, allowing smth' 

"This module enable something"? No that's not right, unless you mean it in the imperative sense, but that doesn't make sense either because we aren't asking the user to do anything.

> 'enables smth and allows smth'

Gramatically correct, but misses the causative relationship between using this module and accepting `Either`

Your original "allowing accepting and returning" was also gramatically correct but I try to avoid two participles like that. It's hard to follow.